src/command.ts

```diff
@@ -9,25 +9,30 @@ export class Command {
     readonly args: string[]
     private readonly cwd?: string
 
-    constructor(program: string, args: string[], cwd?: string) {
-        if (IS_SANDBOXED) {
+    constructor(isSandboxed: boolean, program: string, args: string[], cwd?: string) {
```

Comment:
We can't do the opposite here ?

Reply:
I can't think of way to do the opposite as the other "constructor" has to call the actual constructor

Reply:
I'll see if there is a clean/non hacky way

Reply:
that is what I had in mind 
```typescript
const IS_SANDBOXED: boolean = true;

class Command {
    program: string
    args: string[]

    constructor (program: string, args: string[]) {
        if (IS_SANDBOXED) {
            return Command.sandboxed(program, args)
        } else {
            return Command.host(program, args)
        }
    }

    static sandboxed(program: string, args: string[]): Command {
        args.unshift('--host', '--env=TERM=xterm-256color', program)
        program = 'flatpak-spawn'
        return Command.host(program, args)
    }

    static host(program: string, args: string[]): Command {
        const cmd: Command = {program: program, args: args}
        return cmd
    }
}


const cmd = new Command('flatpak', ['--help'])
console.log(cmd.program)
console.log(cmd.args)
```

Reply:
Typescript doesn't like it as `const cmd: Command = {program: program, args: args}` won't get the same methods of a Command constructed with `new` operator.

Reply:
Actually typescript supports multiple constructors so we could fix this without the need of doing `const cmd: Command = {program: program, args: args}`

Reply:
this could work too, but tests would failed if ran on sandboxed enviroment

```ts
export interface CommandOptions {
    cwd?: string
    forceSandboxed?: boolean
}

export class Command {
    readonly program: string
    readonly args: string[]
    private readonly cwd?: string

    constructor(program: string, args: string[], options?: CommandOptions) {
        if (options?.forceSandboxed || IS_SANDBOXED) {
            this.program = 'flatpak-spawn'
            args.unshift('--host', '--env=TERM=xterm-256color', program)
        } else {
            this.program = program
        }
        this.args = args
        this.cwd = options?.cwd
    }
}
```

Reply:
> this could work too, but tests would failed if ran on sandboxed enviroment
> 
> ```ts
> export interface CommandOptions {
>     cwd?: string
>     forceSandboxed?: boolean
> }
> 
> export class Command {
>     readonly program: string
>     readonly args: string[]
>     private readonly cwd?: string
> 
>     constructor(program: string, args: string[], options?: CommandOptions) {
>         if (options?.forceSandboxed || IS_SANDBOXED) {
>             this.program = 'flatpak-spawn'
>             args.unshift('--host', '--env=TERM=xterm-256color', program)
>         } else {
>             this.program = program
>         }
>         this.args = args
>         this.cwd = options?.cwd
>     }
> }
> ```

That could work indeed, i don't see how you could run the tests in a sandbox env though

Reply:
goes with the CommandOptions one